From b98e1378626bb0c23c31a1c66213c5c33dbab7ae Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Tue, 6 Oct 2015 01:50:25 -0400
Subject: [PATCH 0567/1936] Operation should be done to freq rather than some
 global. arrays freq and oscilFFTfreqs must have refered to the same memory at
 some time in the past, but NOT ANY MORE!  Bwahahaha!

---
 src/Synth/OscilGen.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/Synth/OscilGen.cpp b/src/Synth/OscilGen.cpp
index 4022fa12..4a644cf1 100644
--- a/src/Synth/OscilGen.cpp
+++ b/src/Synth/OscilGen.cpp
@@ -635,6 +635,7 @@ void OscilGen::spectrumadjust(fft_t *freqs)
 {
     if(Psatype == 0)
         return;
+    dprintf(2, "Psapar = %f\n", (float)Psapar);
     float par = Psapar / 127.0f;
     switch(Psatype) {
         case 1:
@@ -652,12 +653,11 @@ void OscilGen::spectrumadjust(fft_t *freqs)
             break;
     }
 
-
     normalize(freqs, synth.oscilsize);
 
     for(int i = 0; i < synth.oscilsize / 2; ++i) {
-        float mag   = abs(oscilFFTfreqs, i);
-        float phase = M_PI_2 - arg(oscilFFTfreqs, i);
+        float mag   = abs(freqs, i);
+        float phase = M_PI_2 - arg(freqs, i);
 
         switch(Psatype) {
             case 1:
-- 
2.47.0

